### ISSUE TITLE
save callback doesn't catch errors

### ISSUE THREAD

**Issue by contra** [neumino/thinky#456]
Reproduce:

```js
const data = {
  id: '123'
}

const inst1 = new Model(data)
const inst2 = new Model(data)

inst1.save((err) => {
  // no error, since this is the first
  
  inst2.save((err) => {
   // error exists, since this is a duplicate
   // notice how we ignore it though and don't log anything
   console.log('Done')
  })
})
```

`Possibly unhandled DuplicatePrimaryKeyError: Duplicate primary key 'id':` magically appears in the logs from who-knows where, even though it was never logged. This type of behavior seems to happen frequently with thinky, where it seems like things just happen without much explanation or documentation.

The only way to prevent this is to add a `.catch` to the end of each save, which means this library doesn't do callbacks very well.

**Comment by neumino** [comment]
Hum, looking at the code now, I don't remember why I was forking promises...
That's probably the main issue here.

**Comment by neumino** [comment]
Fixed in 2.2.5.

I really should rewrite that part one day...
